### PR TITLE
Add ConfigureServices

### PR DESCRIPTION
**Pull request by Stupid-Fox**

### CODE REVIEW

File: CoffeeHouse.Api/Controllers/CoffeeController.cs [hunk 1]
```diff
@@ -14,6 +15,10 @@ public class CoffeeController : ControllerBase
     {
         private readonly IMapper _mapper;
         private readonly ICoffeeService _service;
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddTransient<ICoffeeService, CoffeeService>();
```
Comment (stas1408123) [inline CoffeeHouse.Api/Controllers/CoffeeController.cs]: Add folder Di in BLL 
Implement extension method AddBLLDependencies()

File: CoffeeHouse.BLL/Services/CoffeeService.cs [hunk 2]
```diff
@@ -11,11 +13,15 @@
...
 {
-    internal class CoffeeService : ICoffeeService
+    public class CoffeeService : ICoffeeService
     {
         private readonly IMapper _mapper;
 
         private readonly ICoffeeRepository _repository;
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddTransient<ICoffeeRepository, CoffeeRepository>();
```
Comment (stas1408123) [inline CoffeeHouse.BLL/Services/CoffeeService.cs]: Add folder Di in DAL 
Implement extension method AddDALDependencies()

**Review by stas1408123 — CHANGES_REQUESTED** [review]

**Review by hromau — APPROVED** [review]

**Review by hromau — APPROVED** [review]

File: CoffeeHouse.Api/Program.cs [hunk 3]
```diff
@@ -1,9 +1,13 @@
+using CoffeeHouse.BLL.Services;
+using CoffeeHouse.BLL.Services.Intarfeces;
 using Microsoft.EntityFrameworkCore;
 
 var builder = WebApplication.CreateBuilder();
 
 string connection = builder.Configuration.GetConnectionString("DefaultConnection");
 
+builder.Services.AddSingleton<ICoffeeService, CoffeeService>();
```
Comment (stas1408123) [inline CoffeeHouse.Api/Program.cs]: Remove

File: CoffeeHouse.DAL/Di/Di.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+using System.Threading.Tasks;
+
+namespace CoffeeHouse.DAL.Di
+{
+    public static class Di
+    {
+
+        public static void AddDALDependencies(this IServiceCollection services, IConfiguration config)
+        {
+            services.AddTransient<ICoffeeRepository, CoffeeRepository>();
```
Comment (stas1408123) [inline CoffeeHouse.DAL/Di/Di.cs]: Add database dependencies

**Review by stas1408123 — CHANGES_REQUESTED** [review]

File: CoffeeHouse.DAL/Di/Di.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        }
+        public static IServiceCollection AddDatabase(this IServiceCollection services, IConfiguration configuration)
```
Comment (stas1408123) [inline CoffeeHouse.DAL/Di/Di.cs]: Return void

File: CoffeeHouse.DAL/Di/Di.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        {
+            string connection = configuration.GetConnectionString("DefaultConnection");
+            services.AddDbContext<CoffeeContext>(options => options.UseSqlServer(connection));
```
Comment (stas1408123) [inline CoffeeHouse.DAL/Di/Di.cs]: services.AddDbContext<CoffeeContext>(options => options.UseSqlServer(configuration.GetConnectionString("DefaultConnection")));

File: CoffeeHouse.BLL/Services/CoffeeService.cs [hunk 7]
```diff
@@ -11,12 +16,17 @@
...
-    internal class CoffeeService : ICoffeeService
+    public class CoffeeService : ICoffeeService
     {
         private readonly IMapper _mapper;
 
         private readonly ICoffeeRepository _repository;
 
+        public void ConfigureServices(IServiceCollection services, IConfiguration config)
+        {
+            services.AddDALDependencies(config);
```
Comment (stas1408123) [inline CoffeeHouse.BLL/Services/CoffeeService.cs]: WTF?

File: CoffeeHouse.BLL/Di/Di.cs [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CoffeeHouse.BLL.Di
+{
+    public static class Di
+    {
+       public static void AddBLLDependencies(this IServiceCollection services, IConfiguration config)
+        {
+            services.AddTransient<ICoffeeService, CoffeeService>();
```
Comment (stas1408123) [inline CoffeeHouse.BLL/Di/Di.cs]: Add DAL Dependecy

**Comment by stas1408123** [comment]
Read comments in my past review

**Review by hromau — APPROVED** [review]

**Review by stas1408123 — APPROVED** [review]

**Pull request closed without merging**